src/cromshell/logs/command.py

```diff
@@ -0,0 +1,267 @@
+    "Some standard Cromwell status options are 'ALL', 'Done', 'RetryableFailure', 'Running', and 'Failed'.",
+)
+@click.option(
+    "-p",
+    "--print-logs",
+    is_flag=True,
+    default=False,
+    help="Print the contents of the logs to stdout if true. "
+    "Note: This assumes GCS bucket logs with default permissions otherwise this may not work",
+)
+@click.option(
+    "-des",
+    "--dont-expand-subworkflows",
+    is_flag=True,
+    default=False,
+    help="Do not expand subworkflow info in metadata",
+)
+@click.pass_obj
+def main(
+    config,
+    workflow_id: str,
+    status: list,
+    dont_expand_subworkflows: bool,
+    print_logs: bool,
+):
+    """Get a subset of the workflow metadata."""
+
+    LOGGER.info("logs")
+
+    # If no keys were provided then set key_param to empty else
+    # strip trailing comma from keys and split keys by comma
+    status_param = (
+        ["ALL"]
+        if "ALL".lower() in status.lower()
+        else str(status).strip(",").split(",")
+    )
+
+    metadata_command.check_cromwell_server(config=config, workflow_id=workflow_id)
+
+    LOGGER.info("Status keys set to %s", status_param)
+
+    # To grab the logs we only need a subset of the metadata from the server
+    obtain_and_print_logs(
+        config=config,
+        metadata_param=[
+            "id",
+            "executionStatus",
+            "backendLogs",
+            "subWorkflowMetadata",
+            "subWorkflowId",
+        ],
+        status_params=status_param,
+        dont_expand_subworkflows=dont_expand_subworkflows,
+        cat_logs=print_logs,
+    )
+
+    return 0
+
+
+def obtain_and_print_logs(
+    config,
+    metadata_param: list,
+    cat_logs: bool,
+    status_params: list,
+    dont_expand_subworkflows: bool,
+):
+    """Format metadata parameters and obtains metadata from cromwell server"""
+
+    # Combine keys and flags into a dictionary
+    formatted_metadata_parameter = metadata_command.format_metadata_params(
+        list_of_keys=metadata_param,
+        exclude_keys=False,
+        expand_subworkflows=not dont_expand_subworkflows,  # Invert variable
+    )
+
+    # Request workflow metadata
+    workflow_status_json = metadata_command.get_workflow_metadata(
+        meta_params=formatted_metadata_parameter,
+        api_workflow_id=config.cromwell_api_workflow_id,
+        timeout=config.requests_connect_timeout,
+        verify_certs=config.requests_verify_certs,
+    )
+
+    # Parse the metadata for logs and print them to the output
+    found_logs = print_workflow_logs(
+        workflow_metadata=workflow_status_json,
+        indent="",
+        expand_sub_workflows=not dont_expand_subworkflows,
+        status_keys=status_params,
+        cat_logs=cat_logs,
+    )
+
+    if not found_logs:
+        print(
+            f"No logs with status {status_params} found for workflow, try adding the argument '-s ALL' to list logs with any status"
```

Comment:
Woah, can this format a list automatically? neat!

Reply:
i know python has some cool things that will occasionally catch fire and fail at runtime in utterly unpredictable ways months later...